model.py

```diff
@@ -5504,14 +5512,25 @@ def update_availability(
                 old_value=old_value, new_value=new_value)
 
         # Update the license pool with the latest information.
-        self.licenses_owned = new_licenses_owned
-        self.licenses_available = new_licenses_available
-        self.licenses_reserved = new_licenses_reserved
-        self.patrons_in_hold_queue = new_patrons_in_hold_queue
-        self.last_checked = as_of
+        any_data = False
+        if new_licenses_owned is not None:
+            self.licenses_owned = new_licenses_owned
+            any_data = True
+        if new_licenses_available is not None:
+            self.licenses_available = new_licenses_available
+            any_data = True
+        if new_licenses_reserved is not None:
+            self.licenses_reserved = new_licenses_reserved
+            any_data = True
+        if new_patrons_in_hold_queue is not None:
+            self.patrons_in_hold_queue = new_patrons_in_hold_queue
+            any_data = True
+
+        if changes_made or any_data:
+            self.last_checked = as_of
 
         # Update the last update time of the Work.
-        if self.work:
+        if self.work and (any_data or changes_made):
```

Comment:
minor quibble: checking any_data and changes_made twice.  would it be detrimental to set last_checked outside of a conditional?